File: Treangle/Treangle.ts
import { Figure } from '../Figure/Figure';

/**
 * @param Triangle экспортируем и реализует интерфейс
 * в этом классе находится логика фигуры
 */
export class Triangle implements Figure {
  /**
   * Тип фигуры - треугольник
   */
    type = 'triangle';

    /**
     * конструктор, который принимает три аргумента для сторон треугольника
     * 
     * @param a сторона
     * @param b сторона
     * @param c сторона
     */
    constructor(private a: number, private b: number, private c: number) {}

    /**
     * Метод для вычисления площади треугольника
     */
    public getArea(): number {
      const p = (this.a + this.b + this.c) / 2;
      return Math.sqrt(p * (p - this.a) * (p - this.b) * (p - this.c));
    }

    /**
     *  Метод для вычисления периметра треугольника
     */
    getPerimeter() {
      //периметр треугольника, складывая длины всех его сторон
      return this.a + this.b + this.c;
    }
  }

File: Treangle.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace Canculate_Figur
{
    //Треугольник
    class Treangle:Figure
    {
        private double sideA;
        private double sideB;
        private double sideC;

        public Treangle(double sideA, double sideB, double sideC)
        {
            this.sideA = sideA;
            this.sideB = sideB;
            this.sideC = sideC;
        }

        //Вычисления площади 
        public double GetArea()
        {
            double p = (sideA + sideB + sideC) / 2;
            return Math.Sqrt(p * (p - sideA) * (p - sideB) * (p - sideC));
        }
        //Вычисления периметра 
        public double GetPerimeter()
        {
            return sideA + sideB + sideC;
        }

        //Получения имени фигуры
        public string GetName()
        {
            return "Треугольник";
        }
    }
}
